partiql-parser/src/partiql.pest

```diff
@@ -14,9 +14,39 @@ Scanner = _{ SOI ~ Token }
 Token = _{
       Keyword
     | String
+    | Number
     | Identifier
 }
 
+//
+// Numeric Literals
+//
+
+Number = ${
+      DecimalExp
+    | Decimal
+    | Integer
+}
+
+DecimalExp = {
+    Decimal ~ ("e" | "E") ~ Integer
+}
+
+Decimal = {
+    Integer? ~ "." ~ Fraction
```

Comment:
This will accept some decimals that aren't legal Ion (like `00000.0`, multiple leading zeros) and it will reject others that are legal Ion (like `42.`, no trailing digits). I don't know if PartiQL has different syntax for decimal literals than Ion does or if this is just a first pass that we'll iron out in a future commit, but I wanted to mention it just in case.

Reply:
Yes, according to the SQL spec things like `0000012`, `.001`, `000.001e000010` are all allowed and I have verified that I didn't just port the bad behavior from the original PartiQL parser...

I will add a comment that is explicit about this, also note that I have explicit positive tests for these cases as well.